sdk/lusid/utilities/api_client_builder.py

```diff
@@ -13,62 +13,46 @@
-        api_url = os.getenv("FBN_LUSID_API_URL", None)
-        username = os.getenv("FBN_USERNAME", None)
-        password_raw = os.getenv("FBN_PASSWORD", None)
-        client_id_raw = os.getenv("FBN_CLIENT_ID", None)
-        client_secret_raw = os.getenv("FBN_CLIENT_SECRET", None)
-        app_name = os.getenv("FBN_APP_NAME", "")
-
-        # If any of the environmental variables are missing use a local secrets file
-        if token_url is None or username is None or password_raw is None or client_id_raw is None \
-                or client_secret_raw is None or api_url is None:
-
-            dir_path = os.path.dirname(os.path.realpath(__file__))
-            with open(os.path.join(dir_path, api_configuration), "r") as secrets:
-                config = json.load(secrets)
-
-            token_url = os.getenv("FBN_TOKEN_URL", config["api"]["tokenUrl"])
-            username = os.getenv("FBN_USERNAME", config["api"]["username"])
-            password = pathname2url(os.getenv("FBN_PASSWORD", config["api"]["password"]))
-            client_id = pathname2url(os.getenv("FBN_CLIENT_ID", config["api"]["clientId"]))
-            client_secret = pathname2url(os.getenv("FBN_CLIENT_SECRET", config["api"]["clientSecret"]))
-            api_url = os.getenv("FBN_LUSID_API_URL", config["api"]["apiUrl"])
-            app_name = os.getenv("FBN_APP_NAME", config["api"].get("applicationName", ""))
-
-        else:
-            password = pathname2url(password_raw)
-            client_id = pathname2url(client_id_raw)
-            client_secret = pathname2url(client_secret_raw)
+        # Load the configuration
+        configuration = ApiConfigurationLoader().load(api_configuration)
 
         # Prepare our authentication request
-        token_request_body = ("grant_type=password&username={0}".format(username) +
-                              "&password={0}&scope=openid client groups".format(password) +
-                              "&client_id={0}&client_secret={1}".format(client_id, client_secret))
+        token_request_body = f"grant_type=password&username={configuration.username}" \
+                             f"&password={configuration.password}&scope=openid client groups offline_access" \
+                             f"&client_id={configuration.client_id}&client_secret={configuration.client_secret}"
         headers = {"Accept": "application/json", "Content-Type": "application/x-www-form-urlencoded"}
 
         # Make our authentication request
-        okta_response = requests.post(token_url, data=token_request_body, headers=headers)
+        okta_response = requests.post(configuration.token_url, data=token_request_body, headers=headers)
+
+        if okta_response_handler is not None:
+            okta_response_handler(okta_response)
 
         # Ensure that we have a 200 response code
         assert okta_response.status_code == 200
 
+        okta_json = okta_response.json()
+
         # Retrieve our api token from the authentication response
```

Comment:
Might want to update this comment to describe what is going on.

Reply:
done